docs/algorithms/modular-exponentiation-algorithm.md

```diff
@@ -0,0 +1,82 @@
+---
+id: modular-exponentiation-algorithm
+title: Modular Exponentiation Algorithm
+sidebar_label: Modular Exponentiation
+sidebar_position: 10
+description: "Modular Exponentiation is an algorithm used to efficiently compute large powers modulo a number, using a technique called exponentiation by squaring."
+tags: [Modular Arithmetic, Exponentiation, Algorithms, Number Theory, Recursion]
+---
+
+# Modular Exponentiation Algorithm
+
+## Overview
+**Modular Exponentiation** is an efficient algorithm for computing large powers modulo a number. This algorithm is widely used in cryptography and number theory, where dealing with large numbers is common, and directly calculating powers can lead to overflow. Instead of calculating `base^exp` first and then taking the modulo, **modular exponentiation** computes `(base^exp) % mod` efficiently using the **exponentiation by squaring** method, reducing the time complexity to logarithmic.
+
+## Problem Description
+- **Input**: Three integers, `base`, `exp`, and `mod`, where:
+  - `base` is the base number to be raised to the power.
+  - `exp` is the exponent to which the base is raised.
+  - `mod` is the number by which the result will be reduced.
+- **Output**: The result of `(base^exp) % mod`, calculated efficiently.
+- **Constraints**: Large values for `base` and `exp` make a direct approach infeasible, so modular exponentiation is necessary for efficiency.
+
+## Solution Approach
+The **modular exponentiation algorithm** employs **exponentiation by squaring**, which reduces the problem size at each step by halving the exponent. This approach can be implemented recursively or iteratively and uses the following properties:
+
+- If `exp` is even:  
+  \[
+  \text{base}^{\text{exp}} = (\text{base}^{\text{exp}/2})^2
+  \]
```

Comment:
replace

```md
  \[
  \text{base}^{\text{exp}} = (\text{base}^{\text{exp}/2})^2
  \]
```

with


$$
  \text{base}^{\text{exp}} = (\text{base}^{\text{exp}/2})^2
$$

Reply:
Sure I will make the changes @ajay-dhangar